benchmarks/video/capture_camera_feed.py

```diff
@@ -39,24 +46,21 @@ def display_and_save_video_stream(output_dir: Path, fps: int, width: int, height
     cap.set(cv2.CAP_PROP_FRAME_HEIGHT, height)
 
     frame_index = 0
-    while True:
+    start_time = time.time()
+    while time.time() - start_time < duration:
         ret, frame = cap.read()
 
         if not ret:
             print("Error: Could not read frame.")
             break
-
-        cv2.imshow("Video Stream", frame)
+        rr.log("video/stream", rr.Image(frame.numpy()), static=True)
```

Comment:
I'm curious why you are using `static=True` here? Basically that means Rerun will only show the _latest_ image, with no scroll-back to the earlier images.

If you remove `static=True` you can get video playback. You can also asociate the image with the `frame_index`:

```suggestion
        rr.set_time_sequence("frame_index", frame_index)
        rr.log("video/stream", rr.Image(frame.numpy()))
```

Reply:
Hi @emilk!

We believe that during teleoperation, users don’t necessarily need a full history of all captured images—especially during long sessions, where storing every frame could fill up RAM and eventually truncate the history when memory limits are reached. This approach also aligns with the behavior we previously had with OpenCV, where users only see the latest frame.

That said, we recognize that this doesn’t apply to encoders, where visualizing a plot (and thus retaining historical data) makes more sense for the user experience.